test/distributed/org/apache/cassandra/distributed/test/JMXSslConfigDistributedTest.java

```diff
@@ -0,0 +1,202 @@
+
+import static org.apache.cassandra.config.CassandraRelevantProperties.COM_SUN_MANAGEMENT_JMXREMOTE_SSL;
+import static org.apache.cassandra.config.CassandraRelevantProperties.COM_SUN_MANAGEMENT_JMXREMOTE_SSL_ENABLED_CIPHER_SUITES;
+import static org.apache.cassandra.config.CassandraRelevantProperties.COM_SUN_MANAGEMENT_JMXREMOTE_SSL_ENABLED_PROTOCOLS;
+import static org.apache.cassandra.config.CassandraRelevantProperties.COM_SUN_MANAGEMENT_JMXREMOTE_SSL_NEED_CLIENT_AUTH;
+import static org.apache.cassandra.config.CassandraRelevantProperties.JAVAX_RMI_SSL_CLIENT_ENABLED_CIPHER_SUITES;
+import static org.apache.cassandra.config.CassandraRelevantProperties.JAVAX_RMI_SSL_CLIENT_ENABLED_PROTOCOLS;
+
+/**
+ * Distributed tests for JMX SSL configuration via the system properties OR the encryption options in the cassandra.yaml.
+ */
+public class JMXSslConfigDistributedTest extends AbstractEncryptionOptionsImpl
+{
+    @After
+    public void resetJmxSslSystemProperties()
+    {
+        // The below properties are set as side effect of other properties when tests run. Hence, these are reset here
+        // vs using try-with-resouces block
+        COM_SUN_MANAGEMENT_JMXREMOTE_SSL.reset();
+        COM_SUN_MANAGEMENT_JMXREMOTE_SSL_NEED_CLIENT_AUTH.reset();
+        COM_SUN_MANAGEMENT_JMXREMOTE_SSL_ENABLED_PROTOCOLS.reset();
+        COM_SUN_MANAGEMENT_JMXREMOTE_SSL_ENABLED_CIPHER_SUITES.reset();
+        JAVAX_RMI_SSL_CLIENT_ENABLED_PROTOCOLS.reset();
+        JAVAX_RMI_SSL_CLIENT_ENABLED_CIPHER_SUITES.reset();
+    }
+
+    @Test
+    public void testDefaultEncryptionOptions() throws Throwable
+    {
+        // We must set the keystore in the system variable to make sure that the call to SSLContext.getDefault()
+        // uses it when Client SSL Socketfactory is initialized even if we don't need it here.
+        // The same default SSLContext.getDefault() will be used by other methods like testSystemSettings() in this test
+        // for the Server SSL Socketfactory and at that time we will need the keystore to be available
+        // All of the above is the issue because we run everything (JMX Server, Client) in the same JVM, multiple times
+        // and the SSLContext.getDefault() relies on static initialization that is reused
+        try (WithProperties ignored = new WithProperties().with("javax.net.ssl.trustStore", (String) validKeystore.get("truststore"),
+                                                                "javax.net.ssl.trustStorePassword", (String) validKeystore.get("truststore_password"),
+                                                                "javax.net.ssl.keyStore", (String) validKeystore.get("keystore"),
+                                                                "javax.net.ssl.keyStorePassword", (String) validKeystore.get("keystore_password"))
+        )
+        {
+            ImmutableMap<String, Object> encryptionOptionsMap = ImmutableMap.<String, Object>builder().putAll(validKeystore)
```

Comment:
@maulin-vasavada inline please into try-catch

Reply:
@smiklosovic The challenge is that we have to use this map later in the block within try catch. So I need to have a variable to hold the values unlike the inline setting in the other method. Please check the `line # 90` and you will see what I mean.